CMakeLists.txt

```diff
@@ -156,7 +156,7 @@ install(FILES ${PROJECT_BINARY_DIR}/source/adios2/ADIOSConfig.h
 #------------------------------------------------------------------------------#
 include(CTest)
 mark_as_advanced(BUILD_TESTING)
-add_subdirectory(thirdparty)
+include(thirdparty/dependencies.cmake)
```

Comment:
This should stay add_subdirectory.  Switching to include carries a fair number of scope implications.

Reply:
did you test if `ADIOS2_USE_SYSTEM...` works for the other dependencies? It should not, since `add_subdirectories` only includes newly defined `INTERFACE` libs into the main scope, but not `find_package` modules/config packages: [SO link](https://stackoverflow.com/a/48741931/2719194)